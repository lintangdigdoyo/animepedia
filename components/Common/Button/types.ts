export type ButtonVariantType = "contained" | "outlined";
