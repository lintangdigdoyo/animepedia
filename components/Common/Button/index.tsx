import classNames from "classnames";

import style from "styles/components/Button.module.scss";
import { ButtonVariantType } from "./types";

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  variant?: ButtonVariantType;
}

const Button = ({
  children,
  variant = "contained",
  ...restProps
}: ButtonProps) => {
  return (
    <button
      className={classNames(style.button, {
        [style.button__outlined]: variant === "outlined",
      })}
      {...restProps}
    >
      {children}
    </button>
  );
};

export default Button;
