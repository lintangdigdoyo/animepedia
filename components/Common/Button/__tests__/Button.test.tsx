import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Button from "..";

describe("Button component", () => {
  it("should renders button correctly", () => {
    const props = {
      children: <>children</>,
    };

    const { container } = render(<Button {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should renders button with outlined variant", () => {
    const props = {
      children: <>children</>,
    };

    const { container } = render(<Button variant="outlined" {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should renders button with contained variant", () => {
    const props = {
      children: <>children</>,
    };

    const { container } = render(<Button variant="contained" {...props} />);
    expect(container).toMatchSnapshot();
  });
});
