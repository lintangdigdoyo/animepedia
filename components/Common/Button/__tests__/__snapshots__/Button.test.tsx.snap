// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should renders button correctly 1`] = `
<div>
  <button
    class="button"
  >
    children
  </button>
</div>
`;

exports[`Button component should renders button with contained variant 1`] = `
<div>
  <button
    class="button"
  >
    children
  </button>
</div>
`;

exports[`Button component should renders button with outlined variant 1`] = `
<div>
  <button
    class="button button__outlined"
  >
    children
  </button>
</div>
`;
